repository/src/main/java/org/pentaho/platform/repository2/unified/ExceptionLoggingDecorator.java

```diff
@@ -430,13 +440,18 @@ public RepositoryFileAcl call() throws Exception {
 
   public RepositoryFile updateFile( final RepositoryFile file, final IRepositoryFileData data,
       final String versionMessage ) {
+    Constructor exceptionConstructor = null;
+    try {
+      exceptionConstructor = UnifiedRepositoryUpdateFileException.class.getConstructor( String.class, Throwable.class );
+    } catch ( NoSuchMethodException e ) {
+      e.printStackTrace();
```

Comment:
As above. the e.printStackTrace() essentially kills the exception there, and you go on to use the exceptionConstructor in the return which will be null.